client_servers.go

```diff
@@ -0,0 +1,705 @@
+	Disk        int64  `json:"disk"`
+	IO          int64  `json:"io"`
+	CPU         int64  `json:"cpu"`
+	Threads     string `json:"threads"`
+	OOMDisabled bool   `json:"oom_disabled"`
+}
+
+type FeatureLimits struct {
+	Allocations int `json:"allocations"`
+	Backups     int `json:"backups"`
+	Databases   int `json:"databases"`
+}
+
+type ClientServer struct {
+	ServerOwner bool   `json:"server_owner"`
+	Identifier  string `json:"identifier"`
+	UUID        string `json:"uuid"`
+	InternalID  int    `json:"internal_id"`
+	Name        string `json:"name"`
+	Node        string `json:"node"`
+	SFTP        struct {
+		IP   string `json:"ip"`
+		Port int64  `json:"port"`
+	} `json:"sftp_details"`
+	Description   string        `json:"description"`
+	Limits        Limits        `json:"limits"`
+	Invocation    string        `json:"invocation"`
+	DockerImage   string        `json:"docker_image"`
+	EggFeatures   []string      `json:"egg_features"`
+	FeatureLimits FeatureLimits `json:"feature_limits"`
+	Status        string        `json:"status"`
+	Suspended     bool          `json:"is_suspended"`
+	Installing    bool          `json:"is_installing"`
+	Transferring  bool          `json:"is_transferring"`
+}
+
+func (c *Client) Servers() ([]*ClientServer, error) {
+	req := c.newRequest("GET", "", nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Data []struct {
+			Attributes *ClientServer `json:"attributes"`
+		} `json:"data"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	servers := make([]*ClientServer, 0, len(model.Data))
+	for _, s := range model.Data {
+		servers = append(servers, s.Attributes)
+	}
+
+	return servers, nil
+}
+
+func (c *Client) Server(identifier string) (*ClientServer, error) {
+	req := c.newRequest("GET", "/servers/"+identifier, nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Attributes ClientServer `json:"attributes"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	return &model.Attributes, err
+}
+
+type WebSocketAuth struct {
+	Socket string `json:"socket"`
+	Token  string `json:"token"`
+}
+
+func (c *Client) ServerWebSocket(identifier string) (*WebSocketAuth, error) {
+	req := c.newRequest("GET", fmt.Sprintf("/servers/%s/websocket", identifier), nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Data WebSocketAuth `json:"data"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	return &model.Data, nil
+}
+
+type Resources struct {
+	MemoryBytes    int64 `json:"memory_bytes"`
+	DiskBytes      int64 `json:"disk_bytes"`
+	CPUAbsolute    int64 `json:"cpu_absolute"`
```

Comment:
This should be a `float64` it fails to unmarshal right now.

Reply:
I'm not able to reproduce this 🤔 which field is failing?

Reply:
`CPUAbsolute` is the one that was failing. My CPU usage was `0.014`

I will continue to test when I can.